src/renderers/controls/datetime.control.tsx

```diff
@@ -0,0 +1,62 @@
+import { JSX } from '../JSX';
+import * as _ from 'lodash';
+import { and, formatIs, RankedTester, rankWith, uiTypeIs } from '../../core/testers';
+import { Control, ControlProps, ControlState } from './Control';
+import {
+    formatErrorMessage,
+    mapStateToControlProps,
+    registerStartupRenderer
+} from '../renderer.util';
+import { connect, Event } from '../../common/binding';
+
+/**
+ * Default tester for datetime controls.
+ * @type {RankedTester}
+ */
+export const datetimeControlTester: RankedTester = rankWith(2, and(
+    uiTypeIs('Control'),
+    formatIs('date-time')
+));
+
+export class DateTimeControl extends Control<ControlProps, ControlState> {
+
+    render() {
+        const { classNames, id, visible, enabled, errors, label } = this.props;
+
+        const isValid = errors.length === 0;
+        const divClassNames = 'validation' + (isValid ? '' : ' validation_error');
+
+        const localISOTime = (datestring: string): string => {
```

Comment:
we should make the `localISOTime` method a util, as you need it in the examples and the two renderers.